### PR TITLE
Adding Unit Test for GSLBHostRule

### PR DESCRIPTION
**Pull request by RiyaGaniga-avi**
This PR includes - 

1. Unit test for GSLBHostRule 
2. Addition of mock values to the aviserver to generate fake responses for the tests
3. Code snippet to proportionate the cluster traffic weight of GSLBHostRule object such that they sum to 20

### CODE REVIEW

File: gslb/ingestion/gslb_host_rule.go [hunk 1]
```diff
@@ -76,43 +76,97 @@ func updateGSLBHR(gslbhr *gslbhralphav1.GSLBHostRule, msg string, status string)
...
+			diff += 1
+		}
+	}
+}
+
 func isSitePersistenceProfilePresent(gslbhr *gslbhralphav1.GSLBHostRule, profileName string) bool {
 	// Check if the profile mentioned in gslbHostRule are present as application persistence profile on the gslb leader
 	aviClient := avictrl.SharedAviClients().AviClient[0]
-	uri := "/api/applicationpersistenceprofile?name=" + profileName
+	uri := "api/applicationpersistenceprofile"
```
Comment (hemantavi) [inline gslb/ingestion/gslb_host_rule.go:113]: Is there a reason we are fetching all the app persistence profiles? Shouldn't just the one profile be enough?

File: gslb/ingestion/gslb_host_rule.go [hunk 2]
```diff
@@ -136,7 +190,7 @@ func isThirdPartyMemberSitePresent(gslbhr *gslbhralphav1.GSLBHostRule, siteName
 			}
 		}
 	}
-	gslbutils.Errf("Third Party Member Site %s does not exist", siteName)
+	gslbutils.Warnf("Third Party Member Site %s does not exist", siteName)
```
Comment (hemantavi) [inline gslb/ingestion/gslb_host_rule.go:193]: Let's keep the `Errf` here. As this is an error and not a warning.

File: gslb/ingestion/gslb_host_rule.go [hunk 3]
```diff
@@ -159,21 +212,21 @@ func ValidateGSLBHostRule(gslbhr *gslbhralphav1.GSLBHostRule) error {
...
+			errmsg = "SitePersistence Profile " + sitePersistenceProfileName + " error for " + gslbhrName + " GSLBHostRule"
+			return fmt.Errorf(errmsg)
 		}
 	}
 
 	thirdPartyMembers := gslbhrSpec.ThirdPartyMembers
 	for _, tpmember := range thirdPartyMembers {
 		if vip := net.ParseIP(tpmember.VIP); vip == nil {
 			errmsg := "Invalid VIP for thirdPartyMember site " + tpmember.Site + "," + gslbhrName + " GSLBHostRule (expecting IP address)"
-			updateGSLBHR(gslbhr, errmsg, GslbHostRuleRejected)
```
Comment (hemantavi) [inline gslb/ingestion/gslb_host_rule.go:171]: Shouldn't we update the GSLB HR status if the member is invalid?

Comment (RiyaGaniga-avi) [reply gslb/ingestion/gslb_host_rule.go:171]: Just sending an error to the event handler here. The update of GSLB HR status is taken care of in the event handlers.

**Comment by hemantavi** [comment]
Hey @RiyaGaniga-avi, can we please resolve the merge conflicts?

**Comment by RiyaGaniga-avi** [comment]
> Hey @RiyaGaniga-avi, can we please resolve the merge conflicts?

On it!

File: gslb/ingestion/gslb_host_rule.go (hunk 3 above)
Comment (hemantavi) [reply gslb/ingestion/gslb_host_rule.go:189]: Makes sense.

File: gslb/ingestion/gslb_host_rule.go [hunk 4]
```diff
@@ -200,10 +231,20 @@ func ValidateGSLBHostRule(gslbhr *gslbhralphav1.GSLBHostRule) error {
...
 	}
+
+	trafficsplitElems := gslbhrSpec.TrafficSplit
+	tseSum := 0
+	for _, tse := range trafficsplitElems {
+		tseSum += int(tse.Weight)
+	}
+	if tseSum != 20 {
+		gslbutils.Logf("Sum of weights of cluster traffic weights not equal to 20. Will revise the values to sum up to 20")
+		ProportionateToSum20(gslbhr, tseSum)
```
Comment (hemantavi) [inline gslb/ingestion/gslb_host_rule.go:245]: This call is not required IMO. We can remove this.

**Review by hemantavi — APPROVED** [review]

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**